File: Soal1&Soal2/210015_UTSPBO_Soal1.py
# Nama :Amir Salim
# NPM : 140810210015
# Kelas : A
# Tanggal : 19 Oktober 2022
# Nama Program : Soal1.py
# Deskripsi :  UTS SOAL 1

class Bilangan:
    __bil = int(0)

    #constructor
    def __init__(self):
        self.__bil = 0
    #Method
    def setNum(self,bil):
        self.__bil = bil
    
    def inputNum(self):
        self.__bil = input("Bilangan =  ")
    
    def getNum(self):
        return int(self.__bil);
    
    def outputBil(self):
        print("Bilangan = " , self.__bil)

#-----------------------------------------------------------------------------------------------------------------------------------------

class Perpangkatan:
    __angka=Bilangan()
    __pangkat=Bilangan()

    #constructor
    def __init__(self):
        self.__angka = Bilangan()
        self.__pangkat =Bilangan()
    
    def inputPerpangkatan(self):
        n1 = int(input("Masukkan angka : "))
        n2 = int(input("Masukkan pangkat : "))

        self.__angka.setNum(n1)
        self.__pangkat.setNum(n2)
    
    def hitung(self):
        i = int(1)
        hasil = int(1)

        while(i<=self.__pangkat.getNum()):
            hasil = hasil * self.__angka.getNum()
            i = int(i) + int(1)
        
        return hasil;
    
    def hasil(self):
        print("Pangkat 3 -> (" , self.__angka.getNum() , " , " , self.__pangkat.getNum() , ")" , " = " , self.hitung())
#-----------------------------------------------------------------------------------------------------------------------------------------
p = Perpangkatan()
p.inputPerpangkatan()
p.hasil()

File: Soal1&Soal2/210015_UTSPBO_Soal1.cpp
/*
* Nama :Amir Salim
* NPM : 140810210015
* Kelas : A
* Tanggal : 19 Oktober 2022
* Nama Program : Soal1.cpp
* Deskripsi :  UTS SOAL 1
*/
#include<iostream>

class Bilangan{
    private:
        int bilangan;
    
    public:
        Bilangan(){
            this->bilangan = 0 ;
        }
        Bilangan(int bilangan){
            this->bilangan = bilangan;
        }

        void setNum(int bilangan){
            this->bilangan = bilangan;

        }

        void inputNum(){
            std::cout<<"Bilangan : ";
            std::cin>>this->bilangan;
        }

        int getNum(){
            return this->bilangan;
        }

        void outputBilangan(){
            std::cout<<"Bilangan = "<<this->bilangan;
        }

};

class Perpangkatan{
    private:
        Bilangan angka;
        Bilangan pangkat;
    
    public:
        Perpangkatan(){
            this->angka.setNum(0);
            this->pangkat.setNum(0);
        }

        Perpangkatan(int angka,int pangkat){
            this->angka.setNum(angka);
            this->pangkat.setNum(pangkat);
        }

      

        void inputPerpangkatan(){
            int n1,n2;
            std::cout<<"Masukkan angka = ";
            std::cin>>n1;
            this->angka.setNum(n1);

            std::cout<<"Masukkan pangkat = ";
            std::cin>>n2;
            this->pangkat.setNum(n2);
        }

        int hitungPangkat(){
            int hasil=1;

       
            for(int i=1;i<=this->pangkat.getNum();i++){
                hasil*=this->angka.getNum();
            }
        
        return hasil;
        }

        void outPutPerpangkatan(){
            std::cout<<"Pangkat 3 -> (" << this->angka.getNum()<< " , " << this->pangkat.getNum()<<")" << " = " << this->hitungPangkat();
        }

};

int main(){
    Perpangkatan p1;
    p1.inputPerpangkatan();
    p1.outPutPerpangkatan();
}